actions/emoji_actions.jsx

```diff
@@ -62,3 +63,31 @@ export async function deleteEmoji(emojiId, success, error) {
         error({id: err.server_error_id, ...err});
     }
 }
+
+export function loadRecentlyUsedCustomEmojis() {
+    return async (dispatch, getState) => {
+        if (getState().entities.general.config.EnableCustomEmoji !== 'true') {
+            return {data: true};
+        }
+
+        const recentEmojis = EmojiStore.getRecentEmojis();
+        const emojiMap = getEmojiMap(getState());
+        const missingEmojis = recentEmojis.filter((name) => !emojiMap.has(name));
+
+        missingEmojis.forEach((name) => {
+            EmojiActions.getCustomEmojiByName(name)(dispatch, getState);
```

Comment:
It would be nice if we had a bulk API for this. As it is, we're making up to 27 requests for this

Reply:
I'm not sure I agree, with this there will be caching by both the proxy and the browser. With a bulk request it's going to always go the whole way to the server

Reply:
The ABC team does have a ticket to look into which is really better but using the force I think it's going to be better to have more individual GETs
https://mattermost.atlassian.net/browse/ABC-163